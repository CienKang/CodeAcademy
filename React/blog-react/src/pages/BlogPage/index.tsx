import React from 'react';
import { Footer, Header, Main } from '../../components';

const BlogPage = () :JSX.Element => {
    return ( 
        <div className=''>
            <Header data-testid='header' />
            <Main data-testid='main' />
            <Footer data-testid='footer' />
        </div>
    );
};
 
export default BlogPage;