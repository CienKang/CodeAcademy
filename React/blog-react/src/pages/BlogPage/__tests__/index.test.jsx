import React from 'react';

import { render, screen } from '@testing-library/react';

import BlogPage from '../index';

describe('BlogPage', () => {
    
    it('should render without crashing', () => {
        render(<BlogPage />);
        expect(screen.getByRole('banner')).toBeTruthy();
        expect(screen.getByRole('contentinfo')).toBeTruthy();
        expect(screen.getByRole('main')).toBeTruthy();
    });

    it('should match snapshot', () => {
        const { asFragment } = render(<BlogPage />);
        expect(asFragment()).toMatchSnapshot();
    });
    
});
    