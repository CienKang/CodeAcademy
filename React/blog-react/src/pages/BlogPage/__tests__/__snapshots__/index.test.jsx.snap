// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPage should match snapshot 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <header>
      <div
        class="name-div outer-margin-left-15"
      >
        <h2>
          The Artifact
        </h2>
        <h1>
          <i>
            Culture & Art Blog
          </i>
        </h1>
      </div>
      <div
        class="link-bar-div"
      >
        <span>
          Blog
        </span>
        <span>
          About
        </span>
        <span>
          Contact
        </span>
      </div>
    </header>
    <main>
      <div
        class="cards outer-margin-left-15"
      />
    </main>
    <footer>
      <p
        class="outer-margin-left-15"
      >
        Â©artifact.com 2019
      </p>
    </footer>
  </div>
</DocumentFragment>
`;
